tgui/packages/tgui/interfaces/WeaponVendor/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { toTitleCase } from 'common/string';
+import { useBackend, useLocalState } from '../../backend';
+import { Box, Button, Collapsible, LabeledList, Section, Stack, Table } from '../../components';
+import { Window } from '../../layouts';
+import { WeaponVendorData, WeaponVendorStockData } from './type';
+
+export const pluralize = (word, n) => (n !== 1 ? word + 's' : word);
+
+export const WeaponVendor = (_props, context) => {
+  const { data } = useBackend<WeaponVendorData>(context);
+  const [filterAvailable, setFilterAvailable] = useLocalState(context, 'filter-available', false);
+
+  return (
+    <Window width={550} height={700}>
+      <Window.Content>
+        <Stack className="WeaponVendor" vertical fill>
+          <Stack.Item>
+            <Section fill>
+              <LabeledList>
+                <LabeledList.Item label="Balance">
+                  {Object.entries(data.credits).map(([name, value], index) => (
+                    <Box key={name} inline mr="5px" className={`WeaponVendor__Credits--${name}`}>
+                      {value} {name} {pluralize('credit', value)}
+                      {index + 1 !== Object.keys(data.credits).length ? ', ' : ''}
+                    </Box>
+                  ))}
+                </LabeledList.Item>
+              </LabeledList>
+            </Section>
+          </Stack.Item>
+          <Stack.Item grow={1}>
+            <Section
+              fill
+              scrollable
+              title="Materiel"
+              buttons={
+                <Button.Checkbox checked={filterAvailable} onClick={() => setFilterAvailable(!filterAvailable)}>
+                  Filter Available
+                </Button.Checkbox>
+              }>
+              {Object.keys(data.credits).map((category) => (
+                <StockCategory key={category} category={category} filterAvailable={filterAvailable} />
+              ))}
+            </Section>
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+type StockCategoryProps = {
+  category: string;
+  filterAvailable: boolean;
+};
+
+const StockCategory = ({ category, filterAvailable }: StockCategoryProps, context) => {
+  const { data } = useBackend<WeaponVendorData>(context);
```

Comment:
```suggestion
const StockCategory = (props: StockCategoryProps, context) => {
  const { category, filterAvailable } = props;
  const { data } = useBackend<WeaponVendorData>(context);
```
This isn't enforced by any linting or anything, but I have a strong preference for destructuring props in the function itself rather than in the arguments, it makes maintaining significantly easier.